### DISCUSSION TITLE
API Key not accepted

### DISCUSSION THREAD

**Discussion by j-peeters** [coddingtonbear/obsidian-web#52] [Q&A] [answered]
I have the following problem. When I enter the API key the first time, I get this message:

<img width="500" alt="20221020 Orion Obsidian Web Options@2x" src="https://user-images.githubusercontent.com/4502360/197014743-0d1065e7-d1a6-4f13-bd10-948f56240d49.png">

When I subsequently enter the key in any other Chromium based browser (like Orion, Arc or Vivaldi) I see this.

<img width="500" alt="20221020 Obsidian Web Options" src="https://user-images.githubusercontent.com/4502360/197022027-0d1cbd00-72c3-4e91-a2cc-7ef356bcf37a.png">

What I tried:

- I've copied the API key by clicking on it and cmd-c/cmd-v it into the field. 
- I also pasted it first in a plain text app to make sure no coding was copied. 
- I checked that first and last characters were the same as what I've pasted. 
- I have not typed it manually but with such a long string that is prone to errors.
- I've switched it to insecure connection.
- I have checked that no other apps use the same port on my computer. 
- Lastly I uninstalled and reinstalled the extension in Obsidian to get a fresh API key but that didn't help.

I have been wondering if it is due to the fact that I am on macOS Ventura 13.0 (release candidate) on a MacBook Pro M1Max. But I couldn't think of anything new in Ventura that could prevent this from working.

If anyone has ideas how to solve this I would welcome a reply. Thanks!

**Comment by coddingtonbear** [comment, marked as answer]
@janpeeters70 -- I believe the orange exclamation point is just warning you that it has connected using HTTP instead of HTTPS; if you hover over that shield icon, does it tell you otherwise?

**Reply by j-peeters** [reply]
Owww, I was so preconditioned by the red dot exclamation mark warning that I saw the shield as an error too. It is now indeed connected to the API (via HTTP). Still not able to save a note but I'll ask that in a new thread. I'll mark this one as solved.

**Comment by rtm** [comment]
But what was the original problem, which I assume occurred in Chrome? Is there some Chrome option that needs to be set to make this work?

The odd thing is that I see this in the console:

```
{
    "request": {
        "url": "http://127.0.0.1:27123/",
        "apiKey": "20470ddb1c18eecfec2fd69834752ae3132212a86aae9dd9221b28f18c9e7348",
        "options": {
            "method": "get"
        }
    },
    "response": {
        "status": 200,
        "json": {
            "status": "OK",
            "manifest": {
                "id": "obsidian-local-rest-api",
                "name": "Local REST API",
                "version": "2.5.4",
                "minAppVersion": "0.12.0",
                "description": "Get, change or otherwise interact with your notes in Obsidian via a REST API.",
                "author": "Adam Coddington",
                "authorUrl": "https://coddingtonbear.net/",
                "isDesktopOnly": true,
                "dir": ".obsidian/plugins/obsidian-local-rest-api"
            },
            "versions": {
                "obsidian": "1.6.7",
                "self": "2.5.4"
            },
            "service": "Obsidian Local REST API",
            "authenticated": false
        },
        "text": "{\n  \"status\": \"OK\",\n  \"manifest\": {\n    \"id\": \"obsidian-local-rest-api\",\n    \"name\": \"Local REST API\",\n    \"version\": \"2.5.4\",\n    \"minAppVersion\": \"0.12.0\",\n    \"description\": \"Get, change or otherwise interact with your notes in Obsidian via a REST API.\",\n    \"author\": \"Adam Coddington\",\n    \"authorUrl\": \"https://coddingtonbear.net/\",\n    \"isDesktopOnly\": true,\n    \"dir\": \".obsidian/plugins/obsidian-local-rest-api\"\n  },\n  \"versions\": {\n    \"obsidian\": \"1.6.7\",\n    \"self\": \"2.5.4\"\n  },\n  \"service\": \"Obsidian Local REST API\",\n  \"authenticated\": false\n}"
    }
}
```

**Reply by coddingtonbear** [reply]
That particular message means that the API key you've entered is not correct.

**Comment by rtm** [comment]
![image](https://github.com/user-attachments/assets/60fd1fc3-b587-4d3a-b5a0-61d375dcfe9f)

**Reply by coddingtonbear** [reply]
It's very difficult to say without direct access to your computer, but I honestly can't imagine any alternative -- you're definitely getting a response back from *an* Obsidian, and it's definitely saying to you that your API key is incorrect.  My only off-the-cuff guess may be that you should make sure you're not running Obsidian twice at the same time, but there are probably other alternative possibilities.
